src/Interpreters/SystemLog.cpp

```diff
@@ -65,6 +65,9 @@ std::shared_ptr<TSystemLog> createSystemLog(
         engine = "ENGINE = MergeTree";
         if (!partition_by.empty())
             engine += " PARTITION BY (" + partition_by + ")";
+        int ttl = config.getInt(config_prefix + ".ttl", 0);
```

Comment:
Please also add a check to the above `if`, that if engine is specified manually, ttl must not be specified and must instead go into the engine specification. Like for `<partition_by>`.

Reply:
ok. Will fix here.